core/src/wallet/cosmos/explorers/StargateGaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,1310 @@
+                    !json.GetObject()[kTotal].IsArray()) {
+                    throw make_exception(
+                        api::ErrorCode::HTTP_ERROR,
+                        fmt::format("Failed to get total for {}", endpoint));
+                }
+
+                if (!json.GetObject()[kTotal].IsArray() ||
+                    json.GetObject()[kTotal].GetArray().Size() <= 0) {
+                    return std::make_shared<BigInt>(BigInt::ZERO);
+                }
+
+                for (auto& amount: json.GetObject()[kTotal].GetArray()) {
+                  if (amount.GetObject()[kDenom].GetString() ==
+                      _currency.units[0].code) {
+                    std::string floating_amount =
+                        amount.GetObject()[kAmount].GetString();
+                    floating_amount.erase(floating_amount.find('.'),
+                                          std::string::npos);
+                    return std::make_shared<BigInt>(floating_amount);
+                  }
+                }
+                return std::make_shared<BigInt>(BigInt::ZERO);
+            });
+}
+
+/// Get total unbonding balance
+FuturePtr<BigInt> StargateGaiaCosmosLikeBlockchainExplorer::getUnbondingBalance(
+    const std::string &account) const
+{
+    const auto endpoint = fmt::format(cosmos::constants::kGrpcUnbondingsEndpoint, _ns, _version, account);
+
+    const bool jsonParseNumbersAsString = true;
+
+    return _http->GET(endpoint, ACCEPT_HEADER)
+        .json(jsonParseNumbersAsString)
+        .mapPtr<BigInt>(
+            getContext(),
+            [this, endpoint](const HttpRequest::JsonResult &result) -> std::shared_ptr<BigInt> {
+                auto &json = *std::get<1>(result);
+                if (!json.HasMember(kUnbondingResponses)) {
+                    return std::make_shared<BigInt>(BigInt::ZERO);
+                }
+
+                // Handle null
+                if (!json.GetObject()[kUnbondingResponses].IsArray()) {
+                    return std::make_shared<BigInt>(BigInt::ZERO);
+                }
+                const auto &del_entries = json.GetObject()[kUnbondingResponses].GetArray();
+                BigInt total_amt = BigInt::ZERO;
+                for (const auto &del_val_entries : del_entries) {
+                    const auto &entries = del_val_entries.GetObject()[kEntries].GetArray();
+                    for (const auto &unbondingEntry : entries) {
+                      total_amt =
+                          total_amt + BigInt::fromDecimal(
+                                          unbondingEntry[kBalance].GetString());
+                    }
+                }
+
+                return std::make_shared<BigInt>(total_amt);
+            });
+}
+/// Get total available (spendable) balance
+FuturePtr<BigInt> StargateGaiaCosmosLikeBlockchainExplorer::getSpendableBalance(
+    const std::string &account) const
+{
+    // hardcoded muon
```

Comment:
Not anymore? :)

Reply:
> 2nd and last pass, not many new comments!
> 
> Gaia & Stargate explorers have obviously a lot of similar code that should be put in common, but I assume the Gaia one will be deprecated soon after Stargate is live, so no need to bother if that's the case.

Indeed they should not live together. As we plan to never make this live together I'll just delete the non-Stargate Gaia Explorer and raise an error in the wallet factory if the explorer_engine is still `COSMOS_NODE`